BraveRewardsUI/Pending Contributions/PendingContributionListController.swift

```diff
@@ -0,0 +1,225 @@
+  private lazy var removeAllButton = UIBarButtonItem(title: Strings.removeAllSitesToolbarButtonTitle, style: .plain, target: self, action: #selector(tappedRemoveAll(_:)))
+  private lazy var removeSelectedButton = UIBarButtonItem(title: Strings.removeSelectedToolbarButtonTitle, style: .plain, target: self, action: #selector(tappedRemoveSelected(_:)))
+  
+  private let tableHeaders: [RewardsType: TableHeaderRowView] = [
+    .autoContribute: RewardsType.autoContribute.tableHeaderView,
+    .oneTimeTip: RewardsType.oneTimeTip.tableHeaderView,
+    .recurringTip: RewardsType.recurringTip.tableHeaderView,
+  ]
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    contentView.tableView.delegate = self
+    contentView.tableView.dataSource = self
+    
+    contentView.tableView.allowsMultipleSelectionDuringEditing = true
+    
+    contentView.tableView.register(PendingContributionCell.self)
+    
+    removeSelectedButton.isEnabled = false
+    
+    title = Strings.pendingContributionsTitle.capitalized
+    
+    navigationItem.rightBarButtonItem = editButton
+    toolbarItems = [
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
+      removeAllButton,
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    ]
+  }
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    navigationController?.setToolbarHidden(false, animated: animated)
+    reloadData()
+  }
+  
+  override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    navigationController?.setToolbarHidden(true, animated: animated)
+  }
+  
+  @objc private func tappedEdit() {
+    contentView.tableView.setEditing(true, animated: true)
+    navigationItem.setRightBarButton(doneButton, animated: true)
+    setToolbarItems([
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
+      removeSelectedButton,
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    ], animated: true)
+  }
+  
+  @objc private func tappedDone() {
+    contentView.tableView.setEditing(false, animated: true)
+    navigationItem.setRightBarButton(editButton, animated: true)
+    setToolbarItems([
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
+      removeAllButton,
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    ], animated: true)
+  }
+  
+  // MARK: - Data
+  
+  private var contributionSections: [(type: RewardsType, contributions: [PendingContributionInfo])] = [] {
+    didSet {
+      removeAllButton.isEnabled = !contributionSections.isEmpty
+    }
+  }
+  
+  func reloadData() {
+    state.ledger.pendingContributions { [weak self] list in
+      guard let self = self else { return }
+      self.contributionSections = Dictionary(grouping: list, by: { $0.type })
```

Comment:
Can you add a comment in code about what's happening here?
It's not easy to follow at first look